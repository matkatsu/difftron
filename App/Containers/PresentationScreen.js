// @flow
import React from 'react';
import { connect } from 'react-redux';
import type { Connector } from 'react-redux';
import SplitPane from 'react-split-pane';
import InputLeft from 'Components/InputLeft';
import InputRight from 'Components/InputRight';
import DiffResult from 'Components/DiffResult';
import Actions from 'Actions/Creators';
import type { State, Left, Right, Language, Theme, Format, Split, Contents } from 'Types/State';
import type { Dispatch } from 'Types';

type Props = {
  changeLeft: (input: Left) => void,
  changeRight: (input: Right) => void,
  changeSplit: (event: any) => void,
  left: Left,
  right: Right,
  language: Language,
  theme: Theme,
  contents: Contents,
  format: Format,
  split: Split,
};

export const PresentationScreen = (props: Props) => (
  <SplitPane split="horizontal" defaultSize="60%">
    <SplitPane split="vertical" defaultSize="50%">
      <InputLeft
        onChange={(input: Left) => props.changeLeft(input)}
        value={props.left}
        editorSettings={{ language: props.language, theme: props.theme, format: props.format }}
      />
      <InputRight
        onChange={(input: Right) => props.changeRight(input)}
        value={props.right}
        editorSettings={{ language: props.language, theme: props.theme, format: props.format }}
      />
    </SplitPane>
    <DiffResult
      value={props.contents}
      editorSettings={{
        language: props.language,
        theme: props.theme,
        format: props.format,
        split: props.split,
      }}
      handleSplitChange={e => props.changeSplit(e)}
    />
  </SplitPane>
);

const mapStateToProps = (state: State) => (
  {
    left: state.input.left,
    right: state.input.right,
    language: state.input.language,
    theme: state.input.theme,
    contents: state.output.contents,
    format: state.output.format,
    split: state.output.split,
  }
);

const mapDispatchToProps = (dispatch: Dispatch) => ({
  changeLeft: (input: Left) => {
    dispatch(Actions.inputLeftChange(input));
  },
  changeRight: (input: Right) => {
    dispatch(Actions.inputRightChange(input));
  },
  changeSplit: (event) => {
    dispatch(Actions.outputSplitChange(event.target.value));
  },
});

const connector: Connector<{}, Props> = connect(mapStateToProps, mapDispatchToProps);

export default connector(PresentationScreen);

