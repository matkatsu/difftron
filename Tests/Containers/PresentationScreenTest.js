/* eslint-disable import/first */
import 'ignore-styles';
import '../Setup';
import test from 'ava';
import React from 'react';
import render from 'react-test-renderer';
import { PresentationScreen } from 'Containers/PresentationScreen';

test('PresentationScreen component', (t) => {
  const props = {
    changeLeft: () => ({}),
    changeRight: () => ({}),
    changeSplit: () => ({}),
    left: 'left input',
    right: 'right input',
    language: 'java',
    theme: 'github',
    contents: 'diff contents',
    format: 0,
    split: 'line-by-line',
  };
  const tree = render.create(<PresentationScreen {...props} />).toJSON();
  t.snapshot(tree);
});
