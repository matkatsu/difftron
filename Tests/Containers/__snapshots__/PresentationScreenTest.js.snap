exports[`PresentationScreen component 1`] = `"{\"__ava_react_jsx\":{\"type\":\"div\",\"props\":{\"className\":\"SplitPane  horizontal \",\"style\":{\"display\":\"flex\",\"flex\":1,\"position\":\"absolute\",\"outline\":\"none\",\"overflow\":\"hidden\",\"MozUserSelect\":\"text\",\"WebkitUserSelect\":\"text\",\"msUserSelect\":\"text\",\"userSelect\":\"text\",\"flexDirection\":\"column\",\"height\":\"100%\",\"minHeight\":\"100%\",\"top\":0,\"bottom\":0,\"width\":\"100%\"}},\"children\":[{\"type\":\"div\",\"props\":{\"className\":\"Pane horizontal Pane1\",\"style\":{\"flex\":\"none\",\"position\":\"relative\",\"outline\":\"none\",\"height\":\"60%\",\"display\":\"flex\"}},\"children\":[{\"type\":\"div\",\"props\":{\"className\":\"SplitPane  vertical \",\"style\":{\"display\":\"flex\",\"flex\":1,\"position\":\"absolute\",\"outline\":\"none\",\"overflow\":\"hidden\",\"MozUserSelect\":\"text\",\"WebkitUserSelect\":\"text\",\"msUserSelect\":\"text\",\"userSelect\":\"text\",\"flexDirection\":\"row\",\"height\":\"100%\",\"left\":0,\"right\":0}},\"children\":[{\"type\":\"div\",\"props\":{\"className\":\"Pane vertical Pane1\",\"style\":{\"flex\":\"none\",\"position\":\"relative\",\"outline\":\"none\",\"width\":\"50%\"}},\"children\":[{\"type\":\"div\",\"props\":{\"id\":\"LEFT_INPUT\",\"style\":{\"width\":\"100%\",\"height\":\"100%\"}},\"children\":null}]},{\"type\":\"span\",\"props\":{\"className\":\"Resizer vertical \",\"style\":{}},\"children\":null},{\"type\":\"div\",\"props\":{\"className\":\"Pane vertical Pane2\",\"style\":{\"flex\":1,\"position\":\"relative\",\"outline\":\"none\"}},\"children\":[{\"type\":\"div\",\"props\":{\"id\":\"RIGHT_INPUT\",\"style\":{\"width\":\"100%\",\"height\":\"100%\"}},\"children\":null}]}]}]},{\"type\":\"span\",\"props\":{\"className\":\"Resizer horizontal \",\"style\":{}},\"children\":null},{\"type\":\"div\",\"props\":{\"className\":\"Pane horizontal Pane2\",\"style\":{\"flex\":1,\"position\":\"relative\",\"outline\":\"none\"}},\"children\":[{\"type\":\"div\",\"props\":{\"id\":\"DIFF_RESULT\",\"style\":{\"width\":\"100%\",\"height\":\"100%\"}},\"children\":null}]}]}}"`;
